torch_xla/csrc/device.cpp

```diff
@@ -116,4 +116,16 @@ bool CheckTpuDevice(XlaDeviceType hw_type) {
   return false;
 }
 
+bool CheckNeuronDevice(XlaDeviceType hw_type) {
+  if (hw_type == XlaDeviceType::NEURON) {
+    return true;
+  }
+
+  std::string pjrt_device = runtime::sys_util::GetEnvString("PJRT_DEVICE", "");
+  if (hw_type == XlaDeviceType::SPMD) {
```

Comment:
why check for SPMD, all neuron flow should have tht cast?

Reply:
So just check for pjrt_device == "NEURON"? This preserves the old check and additionally checks for this condition but yes we can only check for the pjrt_device and make that work.